### PR TITLE
[Merged by Bors] - chore(DedekindDomain): use `wlog`

### PR DESCRIPTION
**Pull request by urkud**
* Use `wlog` tactic instead of an explicit `suffices`.
* Restate a lemma in terms of an inequality between fractional ideals.
* Drop an unneeded assumption, thanks to @erdOne 

Co-authored-by: @erdOne 

---
I don't have any specific reasons for this PR. I just saw `WLOG` in the comment that didn't use the `wlog` tactic.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/DedekindDomain/Ideal.lean [hunk 1]
```diff
@@ -392,39 +392,29 @@ namespace FractionalIdeal
...
-    have hM0 := (M.bot_lt_of_maximal hNF).ne'
-    obtain ⟨x, hxM, hx1⟩ := this hM
-    refine ⟨x, inv_anti_mono ?_ ?_ ((coeIdeal_le_coeIdeal _).mpr hIM) hxM, hx1⟩ <;>
-        rw [coeIdeal_ne_zero] <;> assumption
-  -- Let `a` be a nonzero element of `M` and `J` the ideal generated by `a`.
-  intro M hM
-  obtain ⟨⟨a, haM⟩, ha0⟩ := Submodule.nonzero_mem_of_bot_lt (M.bot_lt_of_maximal hNF)
+lemma not_inv_le_one_of_ne_bot [IsDedekindDomain A] (hNF : ¬IsField A) {I : Ideal A}
+    (hI0 : I ≠ ⊥) (hI1 : I ≠ ⊤) : ¬(I⁻¹ : FractionalIdeal A⁰ K) ≤ 1 := by
+  wlog hM : I.IsMaximal generalizing I
```
Comment (erdOne) [inline Mathlib/RingTheory/DedekindDomain/Ideal.lean:397]:
```suggestion
lemma not_inv_le_one_of_ne_bot [IsDedekindDomain A] {I : Ideal A}
    (hI0 : I ≠ ⊥) (hI1 : I ≠ ⊤) : ¬(I⁻¹ : FractionalIdeal A⁰ K) ≤ 1 := by
  have hNF : ¬IsField A := fun h ↦ letI := h.toField; (eq_bot_or_eq_top I).elim hI0 hI1
  wlog hM : I.IsMaximal generalizing I
```
Would you mind doing this in this PR or would you prefer to have it done in a separate one.

File: Mathlib/RingTheory/DedekindDomain/Ideal.lean [hunk 2]
```diff
@@ -471,7 +461,7 @@ theorem mul_inv_cancel_of_le_one [h : IsDedekindDomain A] {I : Ideal A} (hI0 : I
   · rw [← hJ, hJ1, coeIdeal_top]
   obtain ⟨x, hx, hx1⟩ :
     ∃ x : K, x ∈ (J : FractionalIdeal A⁰ K)⁻¹ ∧ x ∉ (1 : FractionalIdeal A⁰ K) :=
-    exists_not_mem_one_of_ne_bot hNF hJ0 hJ1
+    exists_not_mem_one_of_ne_bot hJ0 hJ1
```
Comment (urkud) [inline Mathlib/RingTheory/DedekindDomain/Ideal.lean:464]: I don't know whether we can golf this proof using the `not_inv_le_one_of_ne_bot` version.

**Comment by erdOne** [comment]
I pushed a golf.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**